python/pyspark/sql/connect/dataframe.py

```diff
@@ -249,14 +254,18 @@ def crossJoin(self, other: "DataFrame") -> "DataFrame":
             raise Exception("Cannot cartesian join when self._plan is empty.")
         if other._plan is None:
             raise Exception("Cannot cartesian join when other._plan is empty.")
-
+        self.checkSameSparkSession(other)
         return DataFrame.withPlan(
             plan.Join(left=self._plan, right=other._plan, on=None, how="cross"),
             session=self._session,
         )
 
     crossJoin.__doc__ = PySparkDataFrame.crossJoin.__doc__
 
+    def checkSameSparkSession(self, other: "DataFrame") -> None:
+        if self._session.session_id != other._session.session_id:
+            raise PySparkException("Both Datasets must belong to the same SparkSession")
```

Comment:
I am wondering if a subclass of PySparkException should be used instead of that base class. See https://github.com/apache/spark/blob/master/python/pyspark/errors/exceptions/base.py.

Reply:
Feel free to adjust that in a follow-up though.

Reply:
Thanks for remind, I add `SessionNotSameException` for this. Please check.

Reply:
Thanks! It's better to define an error class in python/pyspark/errors/error_classes.py with a template error message. Again, that's not the blocker of merging the PR.